earth_enterprise/src/fusion/portableglobe/servers/fileunpacker/build_and_test.py

```diff
@@ -25,18 +25,145 @@
+      return False
+    if int(comparand_components[i]) < version_component:
+      return True
+
+  return True
+
+def path_prepend(value, new_path, if_present='skip', separator=':'):
+  """Adds a component to a PATH variable, where paths are separated by
+  `separator`.
+
+    if_present = What to do, if PATH already contains `new_path`.
+      'skip': leave PATH as is.
+      'move': move `new_path` to the beginning of PATH.
+    separator = String that separates path components in the variable value.
+  """
+
+  # Check if the PATH variable already contains `new_path`:
+  if value == new_path:
+    return value
+  if value.startswith(new_path + separator):
+    if if_present == 'skip' or 'move':
+      return value
+    else:
+      raise ValueError('Invalid if_present argument: {0}'.format(if_present))
+  if value.endswith(separator + new_path):
+    if if_present == 'skip':
+      return value
+    elif if_present == 'move':
+      return new_path + separator + value[ : -len(new_path) - 1]
+    else:
+      raise ValueError('Invalid if_present argument: {0}'.format(if_present))
+  i = value.find(separator + new_path + separator)
+  if i >= 0:
+    if if_present == 'skip':
+      return value
+    elif if_present == 'move':
+      return new_path + separator + value[: i] + value[i + len(new_path) + 1:]
+    else:
+      raise ValueError('Invalid if_present argument: {0}'.format(if_present))
+
+  # PATH doesn't contain `new_path`, prepend it:
+  if value == '':
+    return new_path
+
+  return new_path + separator + value
+
+def environ_var_path_prepend(
+  var_name, new_path, if_present='skip', separator=':'
+):
+  """Prepend a path component to a PATH-like environment variable (i.e., one
+  which contains paths separated by `separator`).
+
+    var_name = Name of the environment variable to add a path component to.
+  The variable is created, if it doesn't exist.
+    new_path = Path to add to the variable value.
+    if_present = What to do, if the variable already contains `new_path`.
+      'skip': Leave the variable value as is.
+      'move': Move `new_path` to the beginning of the variable value.
+    separator = String that separates path components in the variable value.
+  """
+
+  try:
+    value = os.environ[var_name]
+  except KeyError:
+    value = ''
+  os.environ[var_name] = path_prepend(
+    value, new_path, if_present=if_present, separator=separator)
+
+def configure_c_compiler(os_dir):
+  if os_dir != 'Linux':
+    return
+  version = get_cc_version('g++')
+  if not version:
+    raise ValueError('Unable to determine g++ version!')
+  if not is_version_ge(version, [4, 8]):
+    # Check for GCC 4.8 from the devtoolset-2-toolchain package on Red Hat 6:
+    cc_dir = '/opt/rh/devtoolset-2/root/usr/bin'
+    if os.path.isfile('{0}/g++'.format(cc_dir)):
+      environ_var_path_prepend('PATH', cc_dir, if_present='move')
+      environ_var_path_prepend(
+        'LIBRARY_PATH',
+        '/opt/rh/devtoolset-2/root/usr/lib',
```

Comment:
Minor: could move most of this path into a shared variable so that it's not repeated.